### ISSUE TITLE
Help understanding how to implement the following.

### ISSUE THREAD

**Issue by Jason-Terry** [microsoft/web-build-tools#1336]
Hey folks, I am trying to learn this framework through building a simple idea of a node-based note taking CLI app. So you can jot quick notes via CLI. Useless I know, but it was something small I thought I could get done.

I am having trouble understanding how to implement the following feature set.

> Usage
> 
> Make a note!
> `note "In package.json a bin property denotes the short hand to be used when installing a package globally"`
> 
> Get the last note
> `note -l`
> 
> Get the all notes
> `note -a`

I have a [repository](https://github.com/Jason-Terry/note-cli-tool) set up if anyone want's to take a look, any help is good help I am really hitting a wall in terms of how I use this library.

**Comment by Jason-Terry** [comment]
Some more information. This is the current outputs when I try the 'note' command.

```
`jason.a.terry@CPX-TGVY1906UUE MINGW64 ~/Personal/Dev/Snips/node-snips/blog-db-api (master)
$ note
Method: NoteAction.onDefineParameters not implemented
usage: note [-h] [-v] -p PAYLOAD <command> ...

Take notes without tabbing out!

Positional arguments:
  <command>
    note                Creates a note in the default notebook

Optional arguments:
  -h, --help            Show this help message and exit.
  -v, --verbose         Show extra logging detail
  -p PAYLOAD, --payload PAYLOAD
                        Required field containing text for the note payload.
                        The default value is "".

For detailed help about a specific command, use: note <command> -h`
```

and with a payload,

```
$ note "This should be a payload"
Method: NoteAction.onDefineParameters not implemented
usage: note [-h] [-v] -p PAYLOAD <command> ...
note: error: argument "<command>": Invalid choice: This should be a payload (choose from [note])
```

**Comment by octogonz** [comment]
I wonder if it would be useful for ts-command-line to include a complete sample that people can refer to when starting a new project. I can see that the existing docs don't really explain the usage very well, and the real world projects we have (Rush, API Documenter, etc) are pretty complicated.

**Comment by Jason-Terry** [comment]
Yeah, if I get this off the ground I think it could be a good explanation project as the scope is rather simple.

I would love to help write those docs / document the process if I could just get a better understanding of the tool itself.

**Comment by octogonz** [comment]
Your syntax of `note "some message"` is not actually supported today. It would need to be a parameter like `note -m "some meesage"`. We could improve this, but the current restrictions were motivated around supporting extensible command lines, e.g. where plugins can contribute extra parameters, so the syntax tries to avoid relying on ordering of operands.

**Comment by Jason-Terry** [comment]
I don't see anything wrong with changing the use case to include the flag, not at all.

It would be nice to have root level commands that can take such arguments, but I think it's fine how you described it.

**Comment by renoirb** [comment]
I agree with this issue.

What if one wanted to provide another utility (say, eslint, or prettier) with a CLI wrapper where we can merge options from the wrapper. Where, we could provide local configuration OR use provided inline.

It's kinda confusing where and how to merge rest (list of strings that aren't `--param paramValue` or `--flag`) and those to be passed up to the wrapped tool. (I'm going to link to a repro).
